Update plugin structure requirements for upload to directory

ioalexei opened the pull request:
Fixes #9192 : 
* add reference to required LICENSE file

Goal: clarify required files for validation in the official plugin repository

Ticket(s): #9192 

- [x] Backport to LTR documentation is requested

DelazJ commented on docs/pyqgis_developer_cookbook/plugins/releasing.rst:
```diff
@@ -102,6 +102,8 @@ a plugin on the official repository:
 #. all required metadata listed in :ref:`metadata table<plugin_metadata_table>`
    must be present
 #. the `version` metadata field must be unique
+#. a licence file must be included, saved as `LICENSE` with no extension (i.e. 
+   not `LICENSE.txt`)
```
```suggestion
#. a licence file must be included, saved as ``LICENSE`` with no extension (i.e. 
   not ``LICENSE.txt`` for example)
```
otherwise no particular formatting in HTML build

DelazJ commented:
@xpirix may I request your feedback here please? Thanks.

Xpirix commented:
Thanks for this change @ioalexei and @DelazJ. This reflects the current plugin upload requirements and will likely help authors successfully upload their plugins.

In the future, the LICENSE requirement may change depending on the discussion at https://github.com/qgis/QGIS-Django/issues/453, but I think this looks good for now.

Gustry commented on docs/pyqgis_developer_cookbook/plugins/releasing.rst:
```diff
@@ -110,10 +112,10 @@ Following the validation rules the compressed (.zip) package of your plugin
-look like.
+files are: :file:`metadata.txt`,  :file:`__init__.py` and :file:`LICENSE`. 
+But it would be nice to have a :file:`README` and of course an icon to represent 
+the plugin (:file:`resources.qrc`). Following is an example of how a plugin.zip 
```
```suggestion
the plugin. Following is an example of how a :file:`plugin.zip` 
```

Gustry reviewed (approved):
Thanks @ioalexei 
It's not the topic in this PR, but I would also remove the resources file. It's mostly not used anymore and I do not recommend it (and also L136, 137)

ioalexei commented:
* Merged suggestions from @DelazJ and @Gustry 
* Fixed a typo
* Added clarification that it applies to updates as well as newly published plugins
* Mentioned the license requirement in the plugin structure setting as well
